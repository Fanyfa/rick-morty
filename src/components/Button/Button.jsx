import { func, string } from 'prop-types';
import React, { Component } from 'react';
import styles from './styles.module.scss';

class Button extends Component {
  static propTypes = {
    text: string.isRequired,
    onClick: func.isRequired,
  };

  render() {
    const { text, onClick } = this.props;
    return (
      <div className={styles.container}>
        <button type="button" onClick={onClick} className={styles.button}>
          {text}
        </button>
      </div>
    );
  }
}

export default Button;
