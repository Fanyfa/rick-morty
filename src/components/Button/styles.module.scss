.container {
  text-align: center;
  .button {
    background-color: #01afc7;
    color: #ffffff;
    text-transform: uppercase;
    font-size: 16px;
    line-height: 1;
    border-radius: 2px;
    background: #01afc7;
    outline: none;
    margin-bottom: 20px;
  }
}
