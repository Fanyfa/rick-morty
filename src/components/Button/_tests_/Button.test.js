import Button from 'components/Button/';
import React from 'react';
import renderer from 'react-test-renderer';

const TEXT = 'Lorem';
const ONCLICK = () => {};

describe('<Button>', () => {
  it('renders', () => {
    const tree = renderer.create(<Button text={TEXT} onClick={ONCLICK} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
