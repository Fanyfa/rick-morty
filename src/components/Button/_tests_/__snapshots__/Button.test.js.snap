// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders 1`] = `
<div
  className="container"
>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    Lorem
  </button>
</div>
`;
